File: qc-core/src/main/java/com/eager/questioncloud/core/domain/coupon/CouponJpaRepository.java
package com.eager.questioncloud.core.domain.coupon;

import java.util.Optional;
import org.springframework.data.jpa.repository.JpaRepository;

public interface CouponJpaRepository extends JpaRepository<CouponEntity, Long> {
    Optional<CouponEntity> findByCode(String code);
}

File: qc-core/qc-pay/qc-pay-rdb/src/main/kotlin/com/eager/questioncloud/payment/repository/CouponJpaRepository.kt
package com.eager.questioncloud.payment.repository

import com.eager.questioncloud.payment.entity.CouponInformationEntity
import org.springframework.data.jpa.repository.JpaRepository
import java.util.*

interface CouponJpaRepository : JpaRepository<CouponInformationEntity, Long> {
    fun findByCode(code: String): Optional<CouponInformationEntity>
}
